File: backend-quarkus/src/main/java/app/nanotes/backend/web/ErrorResponses.java
package app.nanotes.backend.web;

import app.nanotes.backend.web.dto.ErrorDto;
import jakarta.ws.rs.core.MediaType;
import jakarta.ws.rs.core.Response;

public final class ErrorResponses {

    private ErrorResponses() {}

    public static Response of(Response.Status status, String code, String message) {
        return Response.status(status).entity(ErrorDto.of(code, message)).type(MediaType.APPLICATION_JSON).build();
    }
}

File: backend-quarkus/src/main/kotlin/app/nanotes/backend/web/ErrorResponses.kt
package app.nanotes.backend.web

import app.nanotes.backend.web.dto.ErrorDto
import jakarta.ws.rs.core.MediaType
import jakarta.ws.rs.core.Response

object ErrorResponses {
    fun of(status: Response.Status, code: String, message: String): Response =
        Response.status(status).entity(ErrorDto.of(code, message)).type(MediaType.APPLICATION_JSON).build()
}
